OpenRA.Game/Game.cs

```diff
@@ -401,6 +402,54 @@ static string ChooseShellmap()
 		public static void RunAfterTick(Action a) { delayedActions.Add(a); }
 		public static void RunAfterDelay(int delay, Action a) { delayedActions.Add(a, delay); }
 
+		static int pendingScreenshots;
+
+		static void TakeScreenshotInner()
+		{
+			if (Interlocked.Increment(ref pendingScreenshots) < 3)
```

Comment:
Removing it has no adverse effect. My blocking until all OpenGL execution is complete using https://www.opengl.org/sdk/docs/man2/xhtml/glFinish.xml probably solved all the problems this had to workaround before.